x-pack/plugins/code/public/components/admin_page/admin.tsx

```diff
@@ -273,90 +327,100 @@ class AdminPage extends React.PureComponent<Props, State> {
-      <EuiCallOut title={callOutTitle[callOutType!]} color={callOutType} iconType="cross">
-        {callOutMessage}
-      </EuiCallOut>
-    );
-
-    const importRepositoryModal = (
-      <EuiOverlayMask>
-        <EuiModal onClose={this.closeModal} className="importModal">
-          <EuiTitle size="s" className="importModalTitle">
-            <h1>Import Repository</h1>
-          </EuiTitle>
-          <EuiFlexGroup>
-            <EuiFlexItem grow={false}>
-              <EuiSideNav items={items} />
-            </EuiFlexItem>
-            <FlexGrowContainer>
-              <EuiFlexItem className="tabContent">{this.getTabContent()}</EuiFlexItem>
-              {callOut}
-            </FlexGrowContainer>
-          </EuiFlexGroup>
-        </EuiModal>
-      </EuiOverlayMask>
-    );
-
-    const repoList = repos.map(repo => (
-      <RepositoryItem
-        key={repo.uri}
-        repoName={repo.name || ''}
-        repoURI={repo.uri}
-        deleteRepo={this.getDeleteRepoHandler(repo.uri)}
-        indexRepo={this.getIndexRepoHandler(repo.uri)}
-        initRepoCommand={this.props.initRepoCommand.bind(this, repo.uri)}
-        hasInitCmd={this.hasInitCmd(repo)}
-        status={status[repo.uri]}
-        isAdmin={isAdmin}
-      />
-    ));
-
-    return (
-      <EuiPage>
-        <EuiPageBody>
-          <EuiPageContent>
-            <EuiPageContentHeader>
-              <EuiPageContentHeaderSection>
-                <EuiTitle>
-                  <h2>{repositoriesCount} repositories</h2>
-                </EuiTitle>
-              </EuiPageContentHeaderSection>
-              <EuiPageContentHeaderSection>
-                <EuiFlexGroup>
-                  <EuiFlexItem>
-                    <EuiSearchBar className="searchBox" onChange={this.onSearchQueryChange} />
-                  </EuiFlexItem>
-                  <EuiFlexItem grow={false}>
-                    <EuiButtonIcon
-                      className="addRepoButton"
-                      onClick={this.openModal}
-                      iconType="plusInCircle"
-                      aria-label="add"
-                    />
-                  </EuiFlexItem>
-                </EuiFlexGroup>
-              </EuiPageContentHeaderSection>
-            </EuiPageContentHeader>
+  public renderTabContent = () => {
+    switch (this.state.tab) {
+      case AdminTabs.projects: {
+        const repos = this.filterRepos();
+        const repositoriesCount = repos.length;
+        const items = [
+          {
+            name: '',
+            id: '',
+            items: this.getSideNavItems(),
+          },
+        ];
+        const { callOutMessage, status, showCallOut, callOutType, isAdmin } = this.props;
+
+        const callOut = showCallOut && (
+          <EuiCallOut title={callOutTitle[callOutType!]} color={callOutType} iconType="cross">
+            {callOutMessage}
+          </EuiCallOut>
+        );
+
+        const importRepositoryModal = (
```

Comment:
can we break these inlined components into separate files so that it's easier for future style adjustments.